Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs

```diff
@@ -0,0 +1,291 @@
+
+namespace Editors.Audio.Presentation.AudioEditor.ViewModels
+{
+    public class CustomStateDataGridProperties
+    {
+        public string CustomVOActor { get; set; }
+        public string CustomVOCulture { get; set; }
+        public string CustomVOBattleSelection { get; set; }
+        public string CustomVOBattleSpecialAbility { get; set; }
+        public string CustomVOFactionLeader { get; set; }
+    }
+
+    public enum DialogueEventsPreset
+    {
+        None,
+        All,
+        Essential,
+        Custom
+    }
+
+    public partial class AudioEditorViewModel : ObservableObject, IEditorViewModel
+    {
+        public readonly IAudioRepository AudioRepository;
+        public readonly PackFileService PackFileService;
+
+        public string _previousSelectedAudioProjectEvent;
+        [ObservableProperty]
+        private string _selectedAudioProjectEvent;
+        [ObservableProperty]
+        private bool _showCustomStatesOnly;
+
+        // Audio Project settings
+        [ObservableProperty]
+        private string _selectedAudioProjectEventType;
+        [ObservableProperty]
+        private string _selectedAudioProjectEventSubtype;
+        [ObservableProperty]
+        private DialogueEventsPreset _selectedAudioProjectEventsPreset;
+
+        public NotifyAttr<string> DisplayName { get; set; } = new NotifyAttr<string>("Audio Editor");
+        public AudioEditorDataViewModel AudioEditor { get; }
+        public AudioEditorSettingsViewModel AudioEditorSettings { get; }
+
+        // DataGrid observable collections.
+        public ObservableCollection<Dictionary<string, object>> AudioEditorDataGridItems { get; set; } = [];
+        public ObservableCollection<CustomStateDataGridProperties> CustomStatesDataGridItems { get; set; } = [];
+
+        // Data storage for AudioEditorDataGridItems managed in a single instance for ease of access.
+        public static Dictionary<string, List<Dictionary<string, object>>> EventsData => AudioEditorData.Instance.EventsData;
+
+        // Audio Project settings.
+        public List<string> AudioProjectEventType { get; set; } = [];
+        public ObservableCollection<string> AudioProjectSubTypes { get; set; } = [];
+
+        // The list of events in the Audio Project.
+        public ObservableCollection<string> AudioProjectDialogueEvents { get; set; } = [];
+
+        public ICommand AddStatePathCommand { get; set; }
+        public ICommand CreateAudioProjectCommand { get; set; }
+
+        public AudioEditorViewModel(AudioEditorDataViewModel audioEditor, AudioEditorSettingsViewModel audioEditorSettings, IAudioRepository audioRepository, PackFileService packFileService)
+        {
+            AudioEditor = audioEditor;
+            AudioEditorSettings = audioEditorSettings;
+            AudioRepository = audioRepository;
+            PackFileService = packFileService;
+
+            AddStatePathCommand = new RelayCommand(AddStatePath);
```

Comment:
Relay commands can also be cleaned up using the wpf toolkit

Reply:
Oh looks like I missed those ones, good point!